File: medplum-java/medplum-model/src/main/java/com/fhir/Encounter_Participant.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableEncounter_Participant.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Encounter_Participant {
  Optional<List<Extension>> extension();

  Optional<List<Extension>> modifierExtension();

  Optional<List<CodeableConcept>> type();

  Optional<Period> period();

  Optional<Reference> individual();

  Optional<String> id();

  static ImmutableEncounter_Participant.Builder builder() {
    return ImmutableEncounter_Participant.builder();
  }
}

File: medplum-typescript/src/model/fhir/Encounter_Participant.ts
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { Period } from "./Period";
import { Reference } from "./Reference";

export interface Encounter_Participant {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    type: CodeableConcept[] | null | undefined;
    period: Period | null | undefined;
    individual: Reference | null | undefined;
}
